src/MSBuild/XMake.cs

```diff
@@ -1291,6 +1291,43 @@ string outputResultsCache
 
             return success;
         }
+        
+        private static IEnumerable<BuildManager.DeferredBuildMessage> GetMessagesToLogInBuildLoggers()
+        {
+            return new[]
+            {
+                new BuildManager.DeferredBuildMessage(
+                    ResourceUtilities.FormatResourceStringIgnoreCodeAndKeyword(
+                        "Process",
+                        Process.GetCurrentProcess().MainModule?.FileName ?? string.Empty),
```

Comment:
`Process.GetCurrentProcess().MainModule` is IDisposable. I see it doesn't dispose anything, but I guess that it would still go to the finalizer queue (?) if you don't dispose it.

Reply:
maybe @stephentoub knows

Reply:
Process is also disposable and has a finalizer.  In general this pattern for getting the process path is complex and expensive, which is why we introduced Environment.ProcessPath in .NET 6.

Reply:
Yeah my question was is it still necessary to dispose if the dispose method is empty. (I realize in this case it is not empty, the base class one is just useless)

Reply:
For perf, to suppress finalization.